src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -236,7 +236,7 @@ internal static void SerializeAlpnProtocolList(List<SslApplicationProtocol> appl
         internal static unsafe int SslSetAlpnProtos(SafeSslHandle ssl, List<SslApplicationProtocol> applicationProtocols)
         {
             int length = GetAlpnProtocolListSerializedLength(applicationProtocols);
-            Span<byte> buffer = length <= 256 ? stackalloc byte[256].Slice(0, length) : new byte[length];
+            Span<byte> buffer = (uint)length <= 256 ? stackalloc byte[256].Slice(0, length) : new byte[length];
```

Comment:
When are we deciding to do `stackalloc T[const].Slice(0, length)` vs `stackalloc T[length]`?

Reply:
I decided to preserve the current behavior in this PR. Although, we agreed that `stackalloc T[const].Slice(0, length)` pattern is generally better.

Reply:
> Although, we agreed that stackalloc T[const].Slice(0, length) pattern is generally better.

Which is unfortunate as it's not intuitive or canonical: I'd hope we could make "stackalloc T[length]" do the "right thing".

Reply:
> > Although, we agreed that stackalloc T[const].Slice(0, length) pattern is generally better.
> 
> Which is unfortunate as it's not intuitive or canonical: I'd hope we could make "stackalloc T[length]" do the "right thing".

@stephentoub we had a discussion in this PR about it (starting from https://github.com/dotnet/runtime/pull/110864#issuecomment-2557248695 comment and below). 

TL;DR:
`stackalloc T[length]`:
Pros:
- less stack consumption (if the length is less than the threshold)
- simpler code

Cons:
- Performance impact (unknown length leads to additional stack probes, a lot slower initialization if SkipLocalsInit is not set)
- Constant-sized stackalloc reads better/safer

Reply:
Thanks. I understand the _current_ tradeoffs. I'm suggesting we should think through how to avoid the perf const of the simpler approach.

Reply:
> Thanks. I understand the _current_ tradeoffs. I'm suggesting we should think through how to avoid the perf cost of the simpler approach.

I guess performance was not even the main concern in that debate? I agree with Jan that we either want an API that does what is better under the hood, or we invest into escape analysis and just rely on JIT

Reply:
If we can avoid having to write stackalloc altogether, all the better.

But until then, I don't think we should be proliferating changing `stackalloc T[length]` to `stackalloc T[const].Slice(0, length)`. That's just creating less readable / maintainable debt that will also proliferate through the ecosystem as a "best practice" when it's really just a workaround for a perf limitation, and a small one at that. I also disagree with the latter being somehow safer, in particular in all of the relevant cases here, where it's immediately guarded by a length check such that it's obvious just from looking at that one line that the length is in-bounds.